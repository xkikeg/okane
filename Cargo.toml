[workspace]
members = [
    "core",
    "cli",
    "golden",
]

# for workspace, we need to specify resolver explicitly.
resolver = "2"

[workspace.package]
version = "0.16.0-dev"
authors = ["xkikeg"]
edition = "2021"
rust-version = "1.81.0"
license = "MIT"
repository = "https://github.com/xkikeg/okane"

[workspace.dependencies]
# dependencies
bumpalo = {version = "3.17", features = ["collections", "boxed"]}
chrono = "0.4.40"
log = "0.4"
pretty_decimal = { version = "0.2.1", features = ["bounded-static"] }
regex = "1.11"
rust_decimal = "1.37"
strum = { version = "0.27", features = ["derive"] }
thiserror = "2.0"
winnow = "0.7.6"

# middle
env_logger = "0.11"
lazy_static = "1.5"

# dev-dependencies
ctor = "0.4"
indoc = "2.0"
maplit = "1.0"
pretty_assertions = "1.4"
rstest = "0.25"
rust_decimal_macros = "1.37"

[profile.release]
lto = "thin"

[profile.bench]
debug = "limited"
