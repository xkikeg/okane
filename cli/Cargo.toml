[package]
name = "okane"
version.workspace = true
authors.workspace = true
edition.workspace = true
rust-version.workspace = true
license.workspace = true
description = "CLI tool to support Ledger (https://www.ledger-cli.org/) equivalent syntax files."
readme = "../README.md"
repository.workspace = true

[dependencies]
okane-core = { version = "0.14.0-dev", path = "../core" }

# anstream = "0.6"
bumpalo.workspace = true
chrono.workspace = true
clap = { version = "4.5", features = ["derive", "unicode"] }
csv = "1"
either = "1"
encoding_rs = "0.8.35"
encoding_rs_io = "0.1.7"
env_logger.workspace = true
lazy_static.workspace = true
log.workspace = true
path-slash = "0.2.1"
quick-xml = { version = "0.37", features = [ "serialize" ] }
regex = "1"
rust_decimal.workspace = true
serde = { version = "1.0", features = [ "derive" ] }
serde_with = { version = "3.12", features = [ "chrono" ] }
serde_yaml = "0.9.34"
strum.workspace = true
thiserror.workspace = true
winnow.workspace = true

[dev-dependencies]
assert_cmd = "2.0"
# assert_fs = "1.0"
# assert_matches = "1.5"
ctor.workspace = true
indoc.workspace = true
maplit.workspace = true
pretty_assertions.workspace = true
rstest.workspace = true
rust_decimal_macros.workspace = true
shlex = "1.3"
