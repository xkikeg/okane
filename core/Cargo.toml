[package]
name = "okane-core"
version.workspace = true
authors.workspace = true
edition.workspace = true
rust-version.workspace = true
license.workspace = true
description = "Library to support parsing, emitting and processing Ledger (https://www.ledger-cli.org/) format files."
readme = "README.md"
repository.workspace = true

[dependencies]
annotate-snippets = "0.11"
bounded-static = { version = "0.8", default-features = false, features = ["derive", "alloc", "chrono"] }
bumpalo.workspace = true
chrono.workspace = true
derive-where = "1.2"
glob = "0.3.2"
log.workspace = true
rust_decimal.workspace = true
strum = { version = "0.26", features = ["derive"] }
thiserror.workspace = true
unicode-width = "0.2"
winnow.workspace = true

[dev-dependencies]
annotate-snippets = { version = "0.11", features = ["testing-colors"] }
ctor.workspace = true
env_logger.workspace = true
indoc.workspace = true
maplit.workspace = true
pretty_assertions.workspace = true
rstest.workspace = true
rust_decimal_macros.workspace = true
criterion = {version = "0.5", features = ["html_reports"]}

[[bench]]
name = "report_bench"
harness = false

[[bench]]
name = "decimal_bench"
harness = false
