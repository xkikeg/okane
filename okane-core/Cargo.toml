[package]
name = "okane-core"
version.workspace = true
authors.workspace = true
edition.workspace = true
license.workspace = true
description.workspace = true
readme.workspace = true
repository.workspace = true

[dependencies]
chrono = "0.4.24"
log = "0.4"
nom = "7.1"
rust_decimal = "1.28"
strum = { version = "0.24", features = ["derive"] }
thiserror = "1.0"
unicode-width = "0.1.10"

[dev-dependencies]
indoc = "1.0"
maplit = "1.0"
pretty_assertions = "1.3"
rust_decimal_macros = "1.27"
