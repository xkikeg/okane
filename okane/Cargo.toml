[package]
name = "okane"
version.workspace = true
authors.workspace = true
edition.workspace = true
license.workspace = true
description.workspace = true
readme.workspace = true
repository.workspace = true

[dependencies]
okane-core = { path = "../okane-core" }
chrono = "0.4.24"
clap = { version = "4.0", features = ["derive", "unicode"] }
csv = "1"
either = "1"
encoding_rs = "0.8"
encoding_rs_io = "0.1"
env_logger = "0.10"
lazy_static = "1.4"
log = "0.4"
path-slash = "0.2"
quick-xml = { version = "0.27", features = [ "serialize" ] }
regex = "1"
rust_decimal = "1.28"
serde = { version = "1.0", features = [ "derive" ] }
serde_with = { version = "2.2", features = [ "chrono" ] }
serde_yaml = "0.9"
strum = { version = "0.24", features = ["derive"] }
thiserror = "1.0"

[dev-dependencies]
assert_cmd = "2.0"
# # assert_fs = "1.0"
ctor = "0.1"
indoc = "1.0"
maplit = "1.0"
pretty_assertions = "1.3"
rust_decimal_macros = "1.27"
