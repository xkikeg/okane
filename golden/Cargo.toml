[package]
name = "okane-golden"
description = "simple library supporting Golden Testing"
version = "0.1.0"
authors.workspace = true
edition.workspace = true
rust-version.workspace = true
license.workspace = true
repository.workspace = true

[dependencies]
pretty_assertions.workspace = true

[dev-dependencies]
regex.workspace = true
temp-env = "0.3.6"
test-temp-dir = "0.3.3"
